src/engine/SoundManager.java

```diff
@@ -0,0 +1,122 @@
+package engine;
+
+import javax.sound.sampled.*;
+import java.io.*;
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
+ * Manages sound effects and BGM.
+ *
+ * @author <a href="mailto:dpdudyyy@gmail.com">Yun Yeyoung</a>
+ *
+ */
+
+public class SoundManager {
+
+    /** Singleton instance of the class. */
+    private static SoundManager instance;
+    /** Save the sound file **/
+    private static HashMap<String, Clip> soundClips;
+    /** Application logger. */
+    private static Logger logger;
+
+    /**
+     * Private constructor.
+     */
+    private SoundManager() {
+        logger = Core.getLogger();
+        logger.info("Started loading sound resources.");
+
+        soundClips = new HashMap<>();
+
+        try {
+            loadSound("menuBack", "res/sound/SFX/menuBack.wav");
+            loadSound("menuClick", "res/sound/SFX/menuClick.wav");
+            loadSound("menuMove", "res/sound/SFX/menuMove.wav");
+            loadSound("nameTyping", "res/sound/SFX/nameTyping.wav");
+
+            logger.info("Finished loading all sounds.");
+        } catch (IOException e) {
+            logger.warning("Loading failed: IO Exception");
+        } catch (UnsupportedAudioFileException e) {
+            logger.warning("Loading failed: Unsupported audio file: ");
+        } catch (LineUnavailableException e) {
+            logger.warning("Loading failed: Line unavailable: ");
+        }
+    }
+
+    /**
+     * Returns shared instance of SoundManager.
+     *
+     * @return Shared instance of SoundManager.
+     */
+    protected static SoundManager getInstance() {
+        if (instance == null)
+            instance = new SoundManager();
+        return instance;
+    }
+
+    /**
+     * Load the sound and save the map.
+     *
+     * @param soundName Key value of sound
+     * @param filePath Path of the sound file
+     * @throws IOException,UnsupportedAudioFileException,LineUnavailableException exception
+     */
+    public void loadSound(String soundName, String filePath) throws IOException, UnsupportedAudioFileException, LineUnavailableException {
```

Comment:
You can use enum.
The change reason is string key access can occur typo issue.

Please refer this code.
```java
// Sounds.java

public enum Sounds {
    MENU_BACK("menuBack"),
    MENU_CLICK("menuClick"),
    MENU_MOVE("menuMove"),
    MENU_TYPING("menuTyping");
    
    private final String label;
    
    Sounds(String label) {
        this.label = label;
    }
    
    public String getLabel() {
        return this.label;
    }
}

// SoundManager.java

public void loadSound(Sounds sound, String filePath) throws IOException, UnsupportedAudioFileException, LineUnavailableException {
    ...
    soundClips.put(sound, clip)
}

// *Screen.java
SoundManager soundManager = SoundManger.getInstance();
...

soundManager.playSound(Sounds.MENU_BACK);
```